prow/cmd/deck/main.go

```diff
@@ -1140,6 +1148,23 @@ lensesLoop:
 	return viewBuf.String(), nil
 }
 
+func PRHistLink(PRHistLinkTemplate, org, repo string, number int) (string, error) {
+	tmp, err := template.New("t").Parse(PRHistLinkTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed compiling template %q: %v", PRHistLinkTemplate, err)
+	}
+	tmpBuff := bytes.Buffer{}
+	if err = tmp.Execute(&tmpBuff, struct {
```

Comment:
Trying to think of any vulnerabilities... Looks vulnerable to [XSS](https://sucuri.net/guides/what-is-cross-site-scripting/). For example, what happens if I use a template like: `javascript:doSomethingEvil()`

... I think "sanitizing/validating" a template would be tricky because of HTML entities & URL encoding.

I forget if Spyglass configuration is considered  "safe" (requiring PR reviews, plus or minus in-repo config), but I'm assuming two bad actors could collaborate to merge it??

Reply:
Technically this is possible. But this field can only be changed in `config.yaml` file, which is supposed to be guarded by `OWNERS` file, to ensure that changes here can only be merged if a prow admin approve the change. So we might be fine?

Reply:
It's possible that a (new/distracted/etc) OWNER might not pay full attention to it.

I'll leave the final say to you/others.

Reply:
Prow had several other fields like https://github.com/kubernetes/test-infra/pull/24377/files#diff-88e6b42a331a7aa9260f9a830961e3ba3c77e0e6d781465378b510d7abc1517cR997 could also be exposed to this kind of problem. Generally this should be easy to identify during code review